src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,476 @@
+      .extent([[0, 0], [width, height]])
+      .on("zoom", this.zoomFunction);
+
+    const innerSpace = svgViewport
+      .append("g")
+      .attr("width", width)
+      .attr("class", "inner_space")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    innerSpace
+      .append("defs")
+      .append("clipPath")
+      .attr("id", "mask")
+      .append("rect")
+      .attr("width", width)
+      .attr("height", height);
+
+    const graphZone = innerSpace
+      .append("g")
+      .attr("width", width)
+      .attr("height", height)
+      .attr("clip-path", "url(#mask)")
+      .call(zoom);
+
+    graphZone
+      .append("rect")
+      .attr("width", "100%")
+      .attr("height", "100%")
+      .attr("opacity", 0);
+
+    return [innerSpace, graphZone];
+  }
+
+  stackDatas(nameList, valuesArray) {
+    let stackMethod = d3
+      .stack()
+      .keys(nameList)
+      .order(d3.stackOrderNone)
+      .offset(d3.stackOffsetWiggle);
+
+    let stackedData;
+    if (valuesArray.length > 0) {
+      stackedData = stackMethod(valuesArray);
+    }
+    return stackedData;
+  }
+
+  createAndSetAxis(
+    minDate,
+    maxDate,
+    minValue,
+    maxValue,
+    width,
+    height,
+    innerSpace
+  ) {
+    minDate = new Date(String(minDate));
+    maxDate = new Date(String(maxDate));
+
+    this.xAxisScale = d3
+      .scaleTime()
+      .domain([minDate, maxDate])
+      .range([0, width]);
+
+    this.xAxis = d3
+      .axisBottom(this.xAxisScale)
+      .tickFormat(d3.timeFormat("%Y"))
+      .ticks(d3.timeYear, 1);
+
+    this.gx = innerSpace
+      .append("g")
+      .attr("class", "x axis")
+      .attr("transform", "translate(0," + height + ")")
+      .call(this.xAxis);
+
+    this.yAxisScale = d3
+      .scaleLinear()
+      .domain([minValue, maxValue])
+      .range([height, 0]);
+
+    this.yAxisL = d3.axisLeft(this.yAxisScale);
+    this.yAxisR = d3.axisRight(this.yAxisScale);
+
+    this.gyr = innerSpace
+      .append("g")
+      .attr("class", "y axis")
+      .attr("transform", "translate(" + width + ", 0)")
+      .call(this.yAxisR);
+
+    this.gyl = innerSpace
+      .append("g")
+      .attr("class", "y axis")
+      .call(this.yAxisL);
+  }
+
+  createAndSetStreams(layersNumber, graphZone, stackedData, nameList) {
+    let colorNameList = []; // e.g.: [Thematique,#FFFFFF]
+    const z = distinctColors(layersNumber);
+    const area = d3
+      .area()
+      .x((d, i) => {
+        return this.xAxisScale(d.data.date);
+      })
+      .y0(d => {
+        return this.yAxisScale(d[0]);
+      })
+      .y1(d => {
+        return this.yAxisScale(d[1]);
+      });
+
+    this.streams = graphZone
+      .selectAll("path")
+      .data(stackedData)
+      .enter()
+      .append("path")
+      .attr("d", d => {
+        if (d) {
+          return area(d);
+        } else {
+          return [];
+        }
+      })
+      .attr("name", (d, i) => {
+        let name = nameList[i];
```

Comment:
Please, always use `const` by default and only let if you are mutating the value.
But in general, you don't have to mutate anything.

```js
colorNameList.push({ name: nameList[i], color: z[i] });

return name;
```